_posts/2016-10-10-Help!-I'm-becoming-Post-Junior.md

```diff
@@ -0,0 +1,162 @@
+
+## Spit Shine
+
+There is no “One True Solution” for anything in programming. You should be wary of anyone that tries to tell you all similar sounding things should be done one way. The things you work on are likely a combination of different patterns that work together to become “an app.”
+
+Understanding your patterns well, and knowing when to apply them will get you far. If you want to go further still, you need to be able to step back from your patterns and try to see larger pictures. A feature that you write this week  _with a little bit more abstraction today_ could make it much easier to write something next week. 
+
+The hard bit is trying to see what that is. Realistically, this is about understanding where the product is, and what it’s roadmap is for the next few iterations as well as trying to think about abstractions that may not be available inside your project.
+
+## Studying Outside Your Daily Craft
+
+Some problems can be handled particularly well by different architectural patterns.  Without knowing they exist, how can you think that they are something you can use?
+
+This leads to an interesting problem, how do you learn new architectural pattens?
+
+- **Process**: You’re probably using some form of [agile development][3], understand what that [really means][4] and how it compares to others. Try reading [Getting Things Done][5] and I’ve heard good things about [Personal Kanban][6].
+
+-  **Code Architecture**: Here are some book recommendations, [Clean Code][7], [The Pragmatic Programmer][8], [Elements of Reusable Object-Oriented Software][9] and [Working Effectively with Legacy Code][10].
+
+- **Tooling**: There is never one way to do something, so try something else. Switch text editor for a month, or explore alternative methods of doing the same thing.
+
+You can use this knowledge to start offering useful advice that can start to influence your team, for example could be in the form of trying few new idea and offering feedback on their tradeoffs.
+
+## Helping others
+
+You can help your team out by using some of the skills from above.  You can then start thinking of expanding your influence within the company. By being a programmer you already have skills that a lot of people would like to learn or have a better working knowledge of it.
+
+If you’ve got this far in the article, it’s very likely you’ve got enough skills to lead workshops internally, and to write blog posts about your experiences to help other developers at your level. For example, this month I’ve ran a workshop on keyboard shortcuts and on learning to program in Swift. Both of them required maybe 2 hours of preparation, and an email or two announcing that it’s happening.
+
+Other options are to give technical talks within your company on specific topics, for example [Licensing for OSS code][11] is a talk I gave internally to Artsy. We now have a weekly team “Lunch & Learn” where we give anyone the chance to talk, or request a talk on a topic. 
```

Comment:
"are to [give technical talks](http://artsy.github.io/blog/2016/03/09/public-speaking-part1-is-it-for-me/) within your company on" 😄 

Yes, I will write part 3... 😞